### PR TITLE
feat(website): add DocSearch as recommended by docusaurus

### PR DESCRIPTION
**Pull request by s-pace**
👋 team,

I am working on DocSearch. [We have an integration for Docusaurus](https://docusaurus.io/docs/en/search#enabling-the-search-bar). We thought it is a shame you can not also used this search that you can [find on reactjs for example](https://reactjs.org/).

This PR will add DocSearch to the documentation website. It will allow an user to have a learn-as-you-type experience by displaying results thanks to a dropdown in a live way.

Let me know if you need anything.

### CODE REVIEW

**Comment by s-pace** [comment]
Any help needed on this @gustavohenke? You should only need to hit the button 🚀

**Review by gustavohenke — COMMENTED** [review]
@fedeci can you figure out an Algolia API key for us? One that we manage? 
Thanks!

**Comment by fedeci** [comment]
I don't think DocSearch ([docusaurus recommended] (https://docusaurus.io/docs/en/search)) has any options we can handle. I sent a request with my GitHub mail to get an API key, let's see what they say.

**Comment by s-pace** [comment]
Sur you can freely use the following credentilals:
```
apiKey: 'd6d85148b3b81778cf952442d4292bea',
IndexName: 'express-validator'
 ```
 
 Please apply on the docsearch website and we will give you access to the analytics we provide for your account.

**Comment by s-pace** [comment]
You can also just merge this PR and everything will run smoothly on your website

File: website/siteConfig.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 // See https://docusaurus.io/docs/site-config.html for all the possible site configuration options.
 const siteConfig = {
+  algolia: {
+    apiKey: 'd6d85148b3b81778cf952442d4292bea',
+    indexName: 'express-validator'
+  },
```
Comment (fedeci) [inline website/siteConfig.js:6]:
```suggestion
  algolia: {
    apiKey: 'd6d85148b3b81778cf952442d4292bea',
    indexName: 'express-validator',
    algoliaOptions: {
      facetFilters: ['type:lvl3', 'language:LANGUAGE', 'version:VERSION'],
    },
  },
```

**Comment by fedeci** [comment]
This is the same API key I received, so I just added some options.
We can check Algolia configuration [here](https://github.com/algolia/docsearch-configs/blob/master/configs/express-validator.json).

**Comment by s-pace** [comment]
Great. Thank you for the changes. This PR is ready to be merged. You are all set @fedeci

**Comment by fedeci** [comment]
😄Is it possible to send an analytics invitation URL also to @gustavohenke?

**Comment by s-pace** [comment]
Yes sure. DId you apply from https://docsearch.algolia.com/apply/?

**Comment by fedeci** [comment]
Yes

**Pull request merged**